Tests are still failing
`FAILED providers/snowflake/tests/unit/snowflake/hooks/test_snowflake.py::TestPytestSnowflakeHook::test_get_oauth_token - AssertionError: expected call not found.`

--- a/providers/snowflake/src/airflow/providers/snowflake/hooks/snowflake.py
+++ b/providers/snowflake/src/airflow/providers/snowflake/hooks/snowflake.py
@@ -189,8 +189,10 @@ class SnowflakeHook(DbApiHook):
             return extra_dict[field_name] or None
         return extra_dict.get(backcompat_key) or None
 
-    def _get_account_identifier(self, conn_config: dict) -> str:
+    @property
+    def account_identifier(self) -> str:
         """Get snowflake account identifier."""
+        conn_config = self._get_conn_params
         account_identifier = f"https://{conn_config['account']}"
 
         if conn_config["region"]:
@@ -203,8 +205,7 @@ class SnowflakeHook(DbApiHook):
         if conn_config is None:
             conn_config = self._get_conn_params
 
-        account_identifier = self._get_account_identifier(conn_config=conn_config)
-        url = f"{account_identifier}.snowflakecomputing.com/oauth/token-request"
+        url = f"{conn_config['account']}.snowflakecomputing.com/oauth/token-request"
 
         data = {
             "grant_type": "refresh_token",

--- a/providers/snowflake/src/airflow/providers/snowflake/hooks/snowflake.py
+++ b/providers/snowflake/src/airflow/providers/snowflake/hooks/snowflake.py
@@ -189,10 +189,8 @@ class SnowflakeHook(DbApiHook):
             return extra_dict[field_name] or None
         return extra_dict.get(backcompat_key) or None
 
-    @property
-    def account_identifier(self) -> str:
-        """Returns snowflake account identifier."""
-        conn_config = self._get_conn_params
+    def _get_account_identifier(self, conn_config: dict) -> str:
+        """Get snowflake account identifier."""
         account_identifier = f"https://{conn_config['account']}"
 
         if conn_config["region"]:
@@ -205,7 +203,8 @@ class SnowflakeHook(DbApiHook):
         if conn_config is None:
             conn_config = self._get_conn_params
 
-        url = f"{self.account_identifier}.snowflakecomputing.com/oauth/token-request"
+        account_identifier = self._get_account_identifier(conn_config=conn_config)
+        url = f"{account_identifier}.snowflakecomputing.com/oauth/token-request"
 
         data = {
             "grant_type": "refresh_token",

--- a/providers/snowflake/src/airflow/providers/snowflake/hooks/snowflake.py
+++ b/providers/snowflake/src/airflow/providers/snowflake/hooks/snowflake.py
@@ -205,7 +205,7 @@ class SnowflakeHook(DbApiHook):
         if conn_config is None:
             conn_config = self._get_conn_params
 
-        url = f"{conn_config['account']}.snowflakecomputing.com/oauth/token-request"
+        url = f"https://{conn_config['account']}.snowflakecomputing.com/oauth/token-request"
 
         data = {
             "grant_type": "refresh_token",

